pkg/config/viper.go

```diff
@@ -294,7 +294,7 @@ func (c *safeConfig) BindEnv(input ...string) {
 		if c.envKeyReplacer != nil {
 			key = c.envKeyReplacer.Replace(key)
 		}
-		c.configEnvVars = append(c.configEnvVars, key)
+		c.configEnvVars[key] = struct{}{}
```

Comment:
I wonder if throwing a log warning here is desired not only to alert the devs of the conflict but also because we are effectively overwriting the old value. Maybe we only log something if the value is changed?

Reply:
It's overwriting it with the same value -- this is just a set, not a map.  And it's expected that values will already exist, so a warning isn't appropriate.